src/client/component/Markdown.tsx

```diff
@@ -0,0 +1,76 @@
+import { List, Text, Title } from '@mantine/core'
+import type { Root, RootContent } from 'mdast'
+import { useEffect, useState } from 'react'
+import { remark } from 'remark'
+import remarkGfm from 'remark-gfm'
+
+type MarkdownProps = {
+  markdown: string
+}
+
+export function Markdown({ markdown }: MarkdownProps) {
+  const [rootNode, setRootNode] = useState<Root>()
+
+  useEffect(() => {
+    parseMarkdown(markdown).then(setRootNode, console.error)
+  }, [markdown])
+
+  return rootNode ? (
+    <NodeRenderer nodes={rootNode.children} />
+  ) : (
+    <Text c='gray' fs='italic'>
+      読み込みに失敗しました
+    </Text>
+  )
```

Comment:
early return で書くのが好みです
```suggestion
  if (!rootNode) {
    return (
      <Text c='gray' fs='italic'>
        読み込みに失敗しました
      </Text>
    )
  }

  return <NodeRenderer nodes={rootNode.children} />
```